src/groups/metric.jl

```diff
@@ -103,37 +112,39 @@ end
 
 direction(::InvariantMetric{G,D}) where {G,D} = D()
 
-function exp!(M::MetricManifold{<:Manifold,<:InvariantMetric}, ::Val{false}, q, p, X)
-    if has_biinvariant_metric(M) === Val(true)
+function exp!(M::MetricManifold{<:AbstractGroupManifold,<:InvariantMetric}, q, p, X)
+    if has_biinvariant_metric(M)
         conv = direction(metric(M))
-        return retract!(M, q, p, X, GroupExponentialRetraction(conv))
+        return retract!(M.manifold, q, p, X, GroupExponentialRetraction(conv))
```

Comment:
I don't think calling `M.manifold` should be necessary here. A user may want to overload this signature for their specific `MetricManifold`, e.g. to use a custom `MPoint`, and if not, then it should be transparent by default.

Reply:
Sounds reasonable, I will check and change this then.

Reply:
Since the `GroupRetraction`s are only defined for groups, one has to change to something, but the right choice is then of course `group_manifold(M)` which I changed this and the next one to.